src/main/java/com/example/Ecommerce/service/CartServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+package com.example.Ecommerce.service;
+
+import com.example.Ecommerce.dto.CartDto;
+import com.example.Ecommerce.entity.*;
+import com.example.Ecommerce.exception.ResourceNotFoundException;
+import com.example.Ecommerce.mapper.CartMapper;
+import com.example.Ecommerce.repository.CartItemRepo;
+import com.example.Ecommerce.repository.CartRepo;
+import com.example.Ecommerce.repository.ProductRepo;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class CartServiceImpl implements CartService {
+
+    private final CartRepo cartRepo;
+    private final CartItemRepo cartItemRepo;
+    private final UserService userService;
+    private final ProductRepo productRepo;
+
+    @Override
+    public CartDto getCart() {
+        User user = userService.currentUser();
+
+        Cart cart = cartRepo.findByUser(user);
+        if (cart == null) {
+            throw new ResourceNotFoundException("Cart not found for user with name: " + user.getFirstName());
+        }
+        updateCartItemsAndTotalPrice(cart);
```

Comment:
a function with name "getX" should not do any update!

instead, it should do something like

val totalPrice = getCartTotalPrice(cart);
return CartMapper.convertEntityToDto(cart, totalPrice); // so cartDto has totalPrice, but Cart shouldn't